mantle/util/common.go

```diff
@@ -120,3 +122,45 @@ func RunCmdTimeout(timeout time.Duration, cmd string, args ...string) error {
 		return fmt.Errorf("%s timed out after %s", cmd, timeout)
 	}
 }
+
+// ParseDiskSpec converts a disk specification into a Disk. The format is:
+// <size>[:<opt1>,<opt2>,...], like ["5G:channel=nvme"]
+func ParseDiskSpec(spec string) (int64, map[string]string, error) {
+	diskmap := map[string]string{}
+	split := strings.Split(spec, ":")
+	if split[0] == "" || (!strings.HasSuffix(split[0], "G")) {
+		return 0, nil, fmt.Errorf("invalid size opt %s", spec)
+	}
+	var disksize string
+	if len(split) == 1 {
+		disksize = split[0]
+	} else if len(split) == 2 {
+		disksize = split[0]
+		for _, opt := range strings.Split(split[1], ",") {
+			if opt == "mpath" {
+				diskmap["MultiPathDisk"] = "true"
+			} else if opt == "4k" {
+				diskmap["SectorSize"] = "4096"
+			} else if strings.HasPrefix(opt, "channel=") {
+				channel := strings.TrimPrefix(opt, "channel=")
+				if channel == "" {
+					return 0, nil, fmt.Errorf("invalid channel opt %s", opt)
+				}
+				diskmap["Channel"] = channel
+			} else if strings.HasPrefix(opt, "serial=") {
+				serial := strings.TrimPrefix(opt, "serial=")
+				if serial == "" {
+					return 0, nil, fmt.Errorf("invalid serial opt %s", opt)
+				}
+				diskmap["DeviceOpts"] = "serial=" + serial
+			} else {
+				return 0, nil, fmt.Errorf("unknown disk option %s", opt)
+			}
+		}
```

Comment:
What I mean is that this code wouldn't try to look for any specific key at all. It would more be something like (again, semi-pseudocode :)):

```golang
for _, opt := range strings.Split(split[1], ",") {
	kvsplit := strings.Splitn(opt, "=", 2)
	if len(kvsplit) == 0 {
		return fmt.Errorf("invalid empty option found in spec %q", spec)
	} else if len(kvsplit) == 1 {
		diskmap[opt] = ""
	} else {
		diskmap[kvsplit[0]] = kvsplit[1]
	}
}
```